mixin/alerts/compact.libsonnet

```diff
@@ -73,7 +73,7 @@
             annotations: {
               message: 'Thanos Compact {{$labels.job}} has not uploaded anything for 24 hours.',
             },
-            expr: '(time() - max(thanos_objstore_bucket_last_successful_upload_time{%(selector)s})) / 60 / 60 > 24' % thanos.compact,
+            expr: '(time() - max(max_over_time(thanos_objstore_bucket_last_successful_upload_time{%(selector)s}[24h]))) / 60 / 60 > 24' % thanos.compact,
```

Comment:
hmm .. the `max_over_time` doesn't look quite right .. don't we just want to ignore the rollout, so ignore identifying labels? something along the lines of `max without(instance, namespace, pod) (...)` ? (and make identifying labels configurable)

Reply:
Depends what we want to achieve. 

To me we want manual eyes on compactor when it has no uploads for longer time = `1d`. This alert does that, if after 1d none of new instances uploaded anything to bucket, we have a problem (: 

What's missing?

Reply:
Tested, works as expected IMO

Reply:
Ok, got it now. Yes, this seems fine. 
The problem is the fact that for every new deployment the `thanos_objstore_bucket_last_successful_upload_time` until the first upload is `0` so we alert on not having uploaded since 1970... 
However, taking the max_over_time mitigates this. 
LGTM